oidc_provider/tests/cases/test_provider_info_endpoint.py

```diff
@@ -18,12 +24,61 @@ def test_response(self):
         See if the endpoint is returning the corresponding
         server information by checking status, content type, etc.
         """
-        url = reverse('oidc_provider:provider-info')
 
-        request = self.factory.get(url)
+        request = self.factory.get(self.url)
 
         response = ProviderInfoView.as_view()(request)
 
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response['Content-Type'] == 'application/json', True)
         self.assertEqual(bool(response.content), True)
+
+    def test_expected_keys_in_response(self):
+        """
+        Test that response contains all ncecessary keys
+        """
+        required_keys = {
+            'issuer',
+            'authorization_endpoint',
+            'token_endpoint',
+            'userinfo_endpoint',
+            'end_session_endpoint',
+            'introspection_endpoint',
+            'response_types_supported',
+            'jwks_uri',
+            'id_token_signing_alg_values_supported',
+            'subject_types_supported',
+            'token_endpoint_auth_methods_supported',
+            'claims_supported',
+        }
+
+        request = self.factory.get(self.url)
+
+        response = ProviderInfoView.as_view()(request)
+        resp_keys = set(json.loads(response.content.decode('utf-8')).keys())
+        self.assertEqual(required_keys, resp_keys)
```

Comment:
I think the convention here is `assertEqual(observed, expected)` so these should be reversed?

```suggestion
        self.assertEqual(resp_keys, required_keys)
```